jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -140,6 +142,25 @@ private static boolean isProgressFooterEnabled() {
     return System.console() != null && !"dumb".equals(System.getenv("TERM"));
   }
 
+  /**
+   * Gets the major version number from a Java version string.
+   *
+   * <p>Examples: {@code "1.7" -> 7, "1.8.0_161" -> 8, "10" -> 10, "11.0.1" -> 11}
+   *
+   * @param versionString the string to convert
+   * @return the major version number as an integer
+   */
+  private static int getVersionFromString(String versionString) {
```

Comment:
What's the consequence of `versionString` being some nonsense instead of a parseable value?

Would it just be `0`?

Would be nice to add unit tests to this function specifically since it's doing something that is non-trivial.